### PR TITLE
RTEMIS-24: Enabled user email validation.

### PR DESCRIPTION
**Pull request by RohanSingh-inno**
Ticket # : https://innoraft.atlassian.net/browse/RTEMIS-24

---
### What the PR does
`(Explain fixes, changes and impacts)`
- Created a custom module for verifying users email.
- Enabled and tested drupal symfony mailer.

---
### What I have tested
`(Explain usecases you have tested locally)`
- Created a new user with email validation.
- Tested the working of symfony mailer with mailhog.

### Extras
`(Add Screenshots or videos)`
- [Demo Video](https://www.awesomescreenshot.com/video/25270614?key=4fb8c47523f1ead2305f500ee0adbe42)

### CODE REVIEW

File: modules/rte_mis_mail/rte_mis_mail.links.menu.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+rte_mis_mail.Emailsettings:
+  title: Email Confirmation Settings
+  parent: system.admin
+  route_name: rte_mis_mail.Emailsettings
+
+rte_mis_mail.SMSsettings:
```
Comment (kunal-sin) [inline modules/rte_mis_mail/rte_mis_mail.links.menu.yml:6]: `rte_mis_mail.sms.settings`

File: modules/rte_mis_mail/src/Form/EmailSettingsForm.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      '#title' => $this->t('Verification Message'),
+      '#default_value' => $config->get('message'),
+      '#description' => $this->t('The email to send during verification. Replacement parameters for verification code is !code.'),
+      '#states'=>['visible' => [
+        ':input[name="verify"]' => 
+          ['checked' => TRUE],
+        ],
+      ]
+    ];
+    $form['#cache']['tags'] = ['config_evf.settings'];
```
Comment (kunal-sin) [inline modules/rte_mis_mail/src/Form/EmailSettingsForm.php:60]: Why we have added this cache tag?

File: modules/rte_mis_mail/src/Form/EmailSettingsForm.php [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      '#default_value' => $config->get('verify'),
+      '#description' => $this->t('Verification requirement.'),
+    ];
+
+    $form['message'] = [
+      '#type' => 'textarea',
+      '#title' => $this->t('Verification Message'),
+      '#default_value' => $config->get('message'),
+      '#description' => $this->t('The email to send during verification. Replacement parameters for verification code is !code.'),
+      '#states'=>['visible' => [
```
Comment (kunal-sin) [inline modules/rte_mis_mail/src/Form/EmailSettingsForm.php:54]: Please add proper spacing here.

File: modules/rte_mis_mail/src/Form/EmailSettingsForm.php [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+   */
+  protected function getEditableConfigNames() {
+    return ['custom_email_validate.settings'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $config = $this->config('custom_email_validate.settings');
```
Comment (kunal-sin) [inline modules/rte_mis_mail/src/Form/EmailSettingsForm.php:35]: email_form.settings.

File: modules/rte_mis_mail/src/Form/SMSSettingsForm.php [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+  /**
+   * The Mobile Number Utility service.
+   *
+   * @var \Drupal\mobile_number\MobileNumberUtilInterface
+   */
+  protected $mobileNumberUtil;
+
+  /**
+   * Constructs a MobileNumberSettingsForm object.
```
Comment (kunal-sin) [inline modules/rte_mis_mail/src/Form/SMSSettingsForm.php:25]: It should be SmsSettingsForm

File: modules/rte_mis_mail/src/Form/SMSSettingsForm.php [hunk 6]
```diff
@@ -0,0 +1,118 @@
+<?php
```
Comment (kunal-sin) [inline modules/rte_mis_mail/src/Form/SMSSettingsForm.php:1]: Let's change the file name to `SmsSettingsForm`

File: modules/rte_mis_mail/src/Form/SMSSettingsForm.php [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+   */
+  public function getFormId() {
+    return 'mobile_number_config_settings_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return ['custom_sms_settings.settings'];
```
Comment (kunal-sin) [inline modules/rte_mis_mail/src/Form/SMSSettingsForm.php:54]: sms_form.settings

File: modules/rte_mis_mail/src/Form/SMSSettingsForm.php [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $config = $this->config('custom_sms_settings.settings');
+
+    $form['heading'] = [
+      '#type' => 'markup',
+      '#markup' => '<h2>SMS Services</h2>',
+  ];
```
Comment (kunal-sin) [inline modules/rte_mis_mail/src/Form/SMSSettingsForm.php:66]: I don't think it's required. We can add the title directly via the form routing.

File: modules/rte_mis_mail/src/Form/SMSSettingsForm.php [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+      '#description' => $this->t('The SMS message to send during verification.'),
+      '#states'=>['visible' => [
+        ':input[name="verify"]' => 
+          ['checked' => TRUE],
+        ],
+      ]
+    ];
+    
+    return parent::buildForm($form, $form_state);
+  }
```
Comment (kunal-sin) [inline modules/rte_mis_mail/src/Form/SMSSettingsForm.php:87]: Can we combine both Email & SMS settings

File: modules/rte_mis_mail/rte_mis_mail.install [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+/**
+ * @file
+ * Install, update and uninstall functions for this module.
+ */
+
+/**
+ * Implements hook_schema().
+ */
+function rte_mis_mail_schema() {
+  $schema['otp_table'] = [
```
Comment (kunal-sin) [inline modules/rte_mis_mail/rte_mis_mail.install:12]: Change the name to rte_mis_otp

File: modules/rte_mis_mail/src/Form/SMSSettingsForm.php (hunk 9 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_mail/src/Form/SMSSettingsForm.php]: Sure we can.

File: modules/rte_mis_mail/src/Form/SMSSettingsForm.php (hunk 7 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_mail/src/Form/SMSSettingsForm.php]: modified to email_sms_form.settings

File: modules/rte_mis_mail/src/Form/SMSSettingsForm.php (hunk 6 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_mail/src/Form/SMSSettingsForm.php]: modified to email_sms_form.settings

File: modules/rte_mis_mail/rte_mis_mail.links.menu.yml (hunk 1 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_mail/rte_mis_mail.links.menu.yml]: done

File: modules/rte_mis_mail/src/Form/EmailSettingsForm.php (hunk 2 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_mail/src/Form/EmailSettingsForm.php]: Forgot to remove this...was trying something.

File: modules/rte_mis_mail/rte_mis_mail.install (hunk 10 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_mail/rte_mis_mail.install]: Done.

File: modules/rte_mis_mail/src/Form/SMSSettingsForm.php (hunk 5 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_mail/src/Form/SMSSettingsForm.php]: Combined the forms and modified to email_sms_form.settings.

File: modules/rte_mis_mail/src/Form/EmailSettingsForm.php (hunk 4 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_mail/src/Form/EmailSettingsForm.php]: Combined the forms and modified to email_sms_form.settings.

File: modules/rte_mis_mail/src/Form/EmailSettingsForm.php (hunk 3 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_mail/src/Form/EmailSettingsForm.php]: Done.

File: modules/rte_mis_mail/rte_mis_mail.links.menu.yml [hunk 11]
```diff
@@ -0,0 +1,9 @@
+rte_mis_mail.Emailsettings:
```
[the remark this answers is not in the document]
Comment (RohanSingh-inno) [reply modules/rte_mis_mail/rte_mis_mail.links.menu.yml:1]: Combined the forms and modified to email_sms_form.settings.

[the remark this answers is not in the document]
Comment (kunal-sin) [reply modules/rte_mis_mail/rte_mis_mail.links.menu.yml]: It should be like this `rte_mis_mail.email_sms.settings`

File: modules/rte_mis_mail/rte_mis_mail.install [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+        'length' => 255,
+        'not null' => TRUE,
+        'description' => 'Stores on which context is the OTP getting stored.',
+      ],
+
+      'verified' => [
+        'type' => 'int',
+        'size' => 'tiny',
+        'description' => 'Stores the status of OTP matching. 1 for true, 0 for false.',
+        'default' => 0,
```
Comment (kunal-sin) [inline modules/rte_mis_mail/rte_mis_mail.install:46]: I think we should define it as boolean.

File: modules/rte_mis_mail/rte_mis_mail.install [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+      ],
+    ],
+    'primary key' => [
+      'email',
+    ],
```
Comment (kunal-sin) [inline modules/rte_mis_mail/rte_mis_mail.install:51]: Let's add an auto-increment field as well which will also act as a primary key.

File: modules/rte_mis_mail/rte_mis_mail.links.menu.yml [hunk 14]
```diff
@@ -0,0 +1,5 @@
+rte_mis_mail.EmailSmsSettings:
```
Comment (kunal-sin) [inline modules/rte_mis_mail/rte_mis_mail.links.menu.yml:1]: it should be like `rte_mis_mail.email_sms.settings`

File: modules/rte_mis_mail/rte_mis_mail.module [hunk 15]
```diff
@@ -0,0 +1,422 @@
...
+use Drupal\Core\Ajax\AjaxResponse;
+use Drupal\Core\Ajax\HtmlCommand;
+use Drupal\Core\Ajax\ReplaceCommand;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\session_limit\Form\SessionLimitForm;
+use Drupal\Core\Cache\Cache;
+
+/**
+ * Implements hook_form_id_alter.
+ */
```
Comment (kunal-sin) [inline modules/rte_mis_mail/rte_mis_mail.module:17]: Please add the proper hook name here.

File: modules/rte_mis_mail/rte_mis_mail.module [hunk 16]
```diff
@@ -0,0 +1,422 @@
...
+function rte_mis_mail_form_user_register_form_alter(array &$form, FormStateInterface &$form_state) {
+   // Load the configuration object.
+  $config = \Drupal::config('email_sms_form.settings');
+  //  // Get the status of verification
```
Comment (kunal-sin) [inline modules/rte_mis_mail/rte_mis_mail.module:21]: Why we are adding double `//` here?

File: modules/rte_mis_mail/rte_mis_mail.module [hunk 17]
```diff
@@ -0,0 +1,422 @@
...
+  $verify = $config->get('email_verify');
+  $message = $config->get('email_message');
+  $form['#cache']['tags'] = Cache::mergeTags($form['#cache']['tags'], $config->getCacheTags());
+
+  if($verify == TRUE){
+
```
Comment (kunal-sin) [inline modules/rte_mis_mail/rte_mis_mail.module:27]: I'm expecting that we should maintain the proper coding standards here.

File: modules/rte_mis_mail/rte_mis_mail.module [hunk 18]
```diff
@@ -0,0 +1,422 @@
...
+    // Hidden field for controlling the display.
+    $form['display_handler'] = [
+      '#type' => 'hidden',
+      '#default_value' => FALSE,
+    ];
+    // Move the existing email field into the container.
+    $form['email_container']['account']['mail'] = $form['account']['mail'];
+    // Remove the original placement of the email field.
+    unset($form['account']['mail']);
+    // Add the Verify Email button inside the container.
```
Comment (kunal-sin) [inline modules/rte_mis_mail/rte_mis_mail.module:42]: Can't we use `#prefix` & `#suffix` ?

File: modules/rte_mis_mail/rte_mis_mail.module [hunk 19]
```diff
@@ -0,0 +1,422 @@
...
+}
+
+/**
+ * Ajax call for email verify.
+ *
+ * Function to check the email entered, if there persist no error then sending
+ *  the mail,displaying the otp fields and buttons,setting the mail field to
+ *  read only mode, unseting the verify email button.
+ */
+function rte_mis_mail_email_verify(array &$form, FormStateInterface $form_state){
```
Comment (kunal-sin) [inline modules/rte_mis_mail/rte_mis_mail.module:115]: Please add proper spacing in open braces.

File: modules/rte_mis_mail/rte_mis_mail.module [hunk 20]
```diff
@@ -0,0 +1,422 @@
...
+  if (!empty(trim($form_state->getValue('mail'), ""))) {
+    // Call the otp service.
+    $otp_service = \Drupal::service('rte_mis_mail.otp_service');
+    // Calling the email verify function to check for email related errors.
+    $error_msgs = $otp_service->verifyEmail();
+    // Checking if there are any errors.
+    if ($error_msgs != FALSE) {
+      // Get the error message value.
+      $form_state->set('no_cache', TRUE);
+      $response = new AjaxResponse();
```
Comment (kunal-sin) [inline modules/rte_mis_mail/rte_mis_mail.module:125]: Let's define $response object globally, And return at the end.

File: modules/rte_mis_mail/rte_mis_mail.module [hunk 21]
```diff
@@ -0,0 +1,422 @@
...
+  $email = trim($form_state->getValue('mail'), "");
+  if (\Drupal::flood()->isAllowed('otp_resend', 5, 3600, $email)) {
+    if(rte_mis_mail_process_registration_email($email)){
+      $form['email_container']['account']['mail']['#attributes']['readonly'] = 'readonly';
+      unset($form['email_container']['verify_email']);
+      \Drupal::messenger()->deleteAll();
+      \Drupal::flood()->register('otp_resend',3600,$form_state->getValue('mail'));
+      }
+    return $form['email_container'];
+  }else {
```
Comment (kunal-sin) [inline modules/rte_mis_mail/rte_mis_mail.module:173]: Please follow the proper coding standards.

File: modules/rte_mis_mail/rte_mis_mail.module [hunk 22]
```diff
@@ -0,0 +1,422 @@
...
+    // If flood control denies, inform the user.
+    $response = new AjaxResponse();
+    $response->addCommand(new HtmlCommand('#email-validation-message', "You've reached the limit for OTP resend requests. Please try again later."));
```
Comment (kunal-sin) [inline modules/rte_mis_mail/rte_mis_mail.module:176]: The text here should be translatable.

File: modules/rte_mis_mail/rte_mis_mail.module [hunk 23]
```diff
@@ -0,0 +1,422 @@
...
+/**
+ * Function for sending email to user.
+ *
+ * Performs the mail sending process, by generating the otp, checking the
+ * database and sending the mail to the user.
+ *
+ * @param string $email
+ *   The email address of the user.
+ */
+function rte_mis_mail_process_registration_email($email) {
```
Comment (kunal-sin) [inline modules/rte_mis_mail/rte_mis_mail.module:191]: Type is missing for `$email`

File: modules/rte_mis_mail/rte_mis_mail.module [hunk 24]
```diff
@@ -0,0 +1,422 @@
...
+    ];
+  }
+  // Table name where data will be stored.
+  if (($otp_service->checkDatabaseRecord($email)) == FALSE) {
+    $otp_service->insertData($otp_table_values);
+  }
+  else {
+    $otp_service->updateData($otp_table_values);
+  }
+  if($otp_service->generateMail($email, 'rte_mis_mail', 'otp', $otp_table_values['otp'])){
```
Comment (kunal-sin) [inline modules/rte_mis_mail/rte_mis_mail.module:212]: Please follow the proper coding standard

File: modules/rte_mis_mail/rte_mis_mail.module [hunk 25]
```diff
@@ -0,0 +1,422 @@
...
+    return True;
+  }
+  return False;
```
Comment (kunal-sin) [inline modules/rte_mis_mail/rte_mis_mail.module:215]: TRUE or FALSE should always be in caps.

File: modules/rte_mis_mail/rte_mis_mail.module [hunk 26]
```diff
@@ -0,0 +1,422 @@
...
+}
+
+/**
+ * Implements hook_form_alter().
+ */
+function rte_mis_mail_form_alter(&$form, FormStateInterface &$form_state, $form_id) {
+
+  if ($form_id == 'session_limit_form') {
+
+    if ($form_state->get('otp_generated')) {
```
Comment (kunal-sin) [inline modules/rte_mis_mail/rte_mis_mail.module:301]: Can't we combine these two if statements?

File: modules/rte_mis_mail/rte_mis_mail.module (hunk 23 above)
Comment (kunal-sin) [reply modules/rte_mis_mail/rte_mis_mail.module:182]: @RohanSingh-inno This is still pending.

File: modules/rte_mis_mail/rte_mis_mail.module [hunk 27]
```diff
@@ -0,0 +1,411 @@
...
+
+use Drupal\Core\Ajax\AjaxResponse;
+use Drupal\Core\Ajax\HtmlCommand;
+use Drupal\Core\Ajax\ReplaceCommand;
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\session_limit\Form\SessionLimitForm;
+
+/**
+ * Implements hook_form_id_alter().
```
Comment (kunal-sin) [inline modules/rte_mis_mail/rte_mis_mail.module:16]: `Implements hook_form_FORM_ID_alter().`

File: modules/rte_mis_mail/rte_mis_mail.module [hunk 28]
```diff
@@ -0,0 +1,411 @@
...
+ */
+function rte_mis_mail_form_user_register_form_alter(array &$form, FormStateInterface &$form_state) {
+  // Load the configuration object.
+  $config = \Drupal::config('email_sms_form.settings');
+  // Get the status of verification.
+  $verify = $config->get('email_verify');
+  $form['#cache']['tags'] = Cache::mergeTags($form['#cache']['tags'], $config->getCacheTags());
+
+  if ($verify == TRUE) {
```
Comment (kunal-sin) [inline modules/rte_mis_mail/rte_mis_mail.module:25]: Checking `== TRUE` is needed?

File: modules/rte_mis_mail/rte_mis_mail.links.menu.yml [hunk 29]
```diff
@@ -0,0 +1,5 @@
+rte_mis_mail.email_sms.Settings:
+  title: Email & SMS Confirmation Settings
+  parent: system.admin
```
Comment (kunal-sin) [inline modules/rte_mis_mail/rte_mis_mail.links.menu.yml:3]: I don't think we should add it at `system.admin` level. Let's move this inside the configuration menu for now.

File: modules/rte_mis_mail/rte_mis_mail.routing.yml [hunk 30]
```diff
@@ -0,0 +1,8 @@
+rte_mis_mail.EmailSmsSettings:
```
Comment (kunal-sin) [inline modules/rte_mis_mail/rte_mis_mail.routing.yml:1]: `rte_mis_mail.email_sms.settings`

File: modules/rte_mis_mail/rte_mis_mail.module [hunk 31]
```diff
@@ -0,0 +1,411 @@
...
+      $form['submit_otp'] = [
+        '#type' => 'submit',
+        '#value' => t('Submit OTP'),
+        '#submit' => ['rte_mis_mail_submit_otp'],
+        '#validate' => ['rte_mis_mail_validate_otp'],
+      ];
+      $form['resend_otp'] = [
+        '#type' => 'submit',
+        '#value' => t('Resend OTP'),
+        '#submit' => ['rte_mis_mail_resend_otp'],
```
Comment (kunal-sin) [inline modules/rte_mis_mail/rte_mis_mail.module:304]: Don't you think we should do an Ajax call here?

Comment (kunal-sin) [reply modules/rte_mis_mail/rte_mis_mail.module:304]: Also, Can you please check if we get any validation error here when we click on `Resend OTP`?

Validation errors like `UDISE code is required` (if it's empty)

File: modules/rte_mis_mail/rte_mis_mail.module [hunk 32]
```diff
@@ -0,0 +1,411 @@
...
+      ];
+      $form['submit_otp'] = [
+        '#type' => 'submit',
+        '#value' => t('Submit OTP'),
+        '#submit' => ['rte_mis_mail_submit_otp'],
+        '#validate' => ['rte_mis_mail_validate_otp'],
+      ];
+      $form['resend_otp'] = [
+        '#type' => 'submit',
+        '#value' => t('Resend OTP'),
```
Comment (kunal-sin) [inline modules/rte_mis_mail/rte_mis_mail.module:303]: Why we need session limit form check here?

File: modules/rte_mis_mail/rte_mis_mail.module [hunk 33]
```diff
@@ -0,0 +1,411 @@
...
+
+/**
+ * Custom submit handler for the session_limit_form form.
+ */
+function rte_mis_mail_otp_session_limit_form_submit(&$form, &$form_state) {
+
+  if (!$form_state->get('otp_generated')) {
+    rte_mis_mail_generate_otp($form, $form_state);
+  }
+}
```
Comment (kunal-sin) [inline modules/rte_mis_mail/rte_mis_mail.module:320]: Why we need session limit form submit function?

File: modules/rte_mis_mail/rte_mis_mail.module [hunk 34]
```diff
@@ -0,0 +1,411 @@
...
+    // Otp resend.
+    $form['email_container']['otp_container']['otp_resend_button'] = [
+      '#type' => 'button',
+      '#value' => t('Resend OTP'),
+      '#ajax' => [
+        'callback' => 'rte_mis_mail_otp_resend',
+        'wrapper' => 'email-container',
+        'effect' => 'fade',
+      ],
+    ];
```
Comment (kunal-sin) [inline modules/rte_mis_mail/rte_mis_mail.module:86]: Can you please validate on scenario where some required fields are empty and you click resend OTP. Will it throw all the errors or just a message saying OTP sent?

File: modules/rte_mis_mail/rte_mis_mail.links.menu.yml (hunk 29 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_mail/rte_mis_mail.links.menu.yml:3]: ok

File: modules/rte_mis_mail/rte_mis_mail.module (hunk 27 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_mail/rte_mis_mail.module:16]: Ok.

File: modules/rte_mis_mail/rte_mis_mail.routing.yml (hunk 30 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_mail/rte_mis_mail.routing.yml:1]: ok

File: modules/rte_mis_mail/rte_mis_mail.module (hunk 34 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_mail/rte_mis_mail.module:85]: Sure. I have validated the above scenario and its working fine.

File: modules/rte_mis_mail/rte_mis_mail.install (hunk 12 above)
Comment (kunal-sin) [reply modules/rte_mis_mail/rte_mis_mail.install:53]: Let's add the following as well:

'unsigned' => TRUE,
'not null' => TRUE,

File: modules/rte_mis_mail/rte_mis_mail.module (hunk 33 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_mail/rte_mis_mail.module]: Thought of different section and measure, but I don't think we need it here.Hence, removed that.

File: modules/rte_mis_mail/js/rte_mis_mail.js [hunk 35]
```diff
@@ -0,0 +1,45 @@
+(function ($, Drupal, once) {
+  Drupal.behaviors.customBehavior = {
+    attach: function (context, settings) {
+
+      if (settings['otp_prompt']) {
+        $('#otp-container').addClass('show').removeClass('hide');
+      }
+      else {
+        $('#otp-container').addClass('hide').removeClass('show');
+      }
```
Comment (kunal-sin) [inline modules/rte_mis_mail/js/rte_mis_mail.js:10]: Please add proper comment

File: modules/rte_mis_mail/js/rte_mis_mail.js [hunk 36]
```diff
@@ -0,0 +1,45 @@
...
+            container.find('#otp-container input[type="number"]').val('');
+            container.find('#otp-container').addClass('hide').removeClass('show');
+            container.find('.form-item.verified').removeClass('show').addClass('hide');
+          }
+        });
+      })
+
+      $(once('otp-container', 'body', context)).each(function () {
+        $('#otp-container').addClass('hide').removeClass('show');
+      })
```
Comment (kunal-sin) [inline modules/rte_mis_mail/js/rte_mis_mail.js:42]: Add comment pls

File: modules/rte_mis_mail/rte_mis_mail.module [hunk 37]
```diff
@@ -0,0 +1,361 @@
...
+use Drupal\Core\Ajax\SettingsCommand;
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\rte_mis_mail\Services\OtpService;
+use Drupal\user\UserInterface;
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function rte_mis_mail_form_user_form_alter(&$form, $form_state, $x) {
```
Comment (kunal-sin) [inline modules/rte_mis_mail/rte_mis_mail.module:20]: What is this `$x`?

File: modules/rte_mis_mail/js/rte_mis_mail.js (hunk 35 above)
Comment (jaswinsingh) [reply modules/rte_mis_mail/js/rte_mis_mail.js]: Added

File: modules/rte_mis_mail/js/rte_mis_mail.js (hunk 36 above)
Comment (jaswinsingh) [reply modules/rte_mis_mail/js/rte_mis_mail.js]: Added

File: modules/rte_mis_mail/rte_mis_mail.module (hunk 37 above)
Comment (jaswinsingh) [reply modules/rte_mis_mail/rte_mis_mail.module]: Was checking the form ID, removed it.

**Review by kunal-sin — APPROVED** [review]

**Pull request merged**